tests/TestCase/ORM/RulesCheckerIntegrationTest.php

```diff
@@ -370,6 +370,58 @@ public function testIsUniqueMultipleFields()
     }
 
     /**
+     * Tests isUnique with checkNull
+     *
+     * @group save
+     * @return void
+     */
+    public function testIsUniqueCheckNull()
+    {
+        $entity = new Entity([
+            'log_id' => null
+        ]);
+
+        $table = TableRegistry::get('Logs');
+        $rules = $table->rulesChecker();
+        $rules->add($rules->isUnique(['log_id']), ['checkNull' => true]);
```

Comment:
If you're re-using an existing feature you could insert a row with the null field before trying to duplicate it and trigger the new option working.